### ISSUE TITLE
sparse placeholder initialization fails due to tensorflow bug

### ISSUE THREAD

**Issue by michaelosthege** [fchollet/keras#5225]
~~Sparse placeholder initialization by `K.placeholder` does not specify the `shape` parameter in either [tensorflow](https://github.com/fchollet/keras/blob/master/keras/backend/tensorflow_backend.py#L309) or [theano](https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py#L108) backends.~~

~~According to the [tensorflow API docs](https://www.tensorflow.org/api_docs/python/io_ops/placeholders#sparse_placeholder) the `shape` parameter is optional. (If the shape is not specified, you can feed a sparse tensor of any shape.)
But since Tensor shapes are tracked (through the _keras_shape) attribute, this leads to problems in the computation graph.
I noticed this when `K.sum(t_sparse*t_dense)` in a metric lead to unexpected behaviour.~~

The solution to this issue:

[tensorflow_backend.py:347](https://github.com/fchollet/keras/blob/master/keras/backend/tensorflow_backend.py#L347) ~~and [theano_backend.py:182](https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py#L182)~~
```python
tf.sparse_placeholder(dtype, name=name, shape=numpy.array(shape).astype("int64"))
```
Note that the `numpy.array` type conversion to `int64` is a workaround for [tensorflow issue #6749](https://github.com/tensorflow/tensorflow/issues/6749)

Note that the theano backend does not specify a shape/broadcast parameter.
~~Was the `shape` left out on purpose? If not I'll open a PR to fix it.~~

cheers

**Comment by michaelosthege** [comment]
More details:
Any Tensor that results from an operation on the sparse placeholder of unknown (None) shape, also has None-dimensions in its shape.
In my case this leads to memory leaks of some sort when doing K.sum(a_tensor_with_none_shape).

**Comment by michaelosthege** [comment]
I have opened PR #5773 with a workaround until [tensorflow issue #6749](https://github.com/tensorflow/tensorflow/issues/6749) is resolved
